compiler/luci/lang/include/luci/IR/CircleQuantParam.h

```diff
@@ -28,6 +28,7 @@ struct CircleQuantParam
   std::vector<float> min;
   std::vector<float> max;
   std::vector<float> scale;
+  std::vector<float> hessian;
```

Comment:
This change means that all `CircleQuantParam` instances have `hessian`, which would have a significant impact.

Is `hessian` computed for all `CircleQuantParam` instances? Or is it just for some tensors? Isn't is possible to save hessian to somewhere else (to decouple it from existing struct)?

Reply:
```hessian``` is only valid for the input tensors of layers that have weights. Could there be a better way?

Reply:
I skimmed through #13585. It seems that `quantize_dequantize_weights_with_gptq` option runs two modules `RecordHessian` and `Quantize weights with GPTQ`. Input/output of those modules are as follows.

```
RecordHessian
- Input
  - module, input_data
- Output
  - module (hessian embedded)

Quantize weights with GPTQ
- Input
  - module (hessian embedded)
- Output
  - module (quantized)
```

Could you change input/output of those modules as below?

```
RecordHessian
- Input
  - module, input_data
- Output
  - hessians: Map(node->vector)

Quantize weights with GPTQ
- Input
  - module, hessians
- Output
  - module (quantized)
```

I think hessian information can be passed separately (`hessians`), rather than saved in `CircleQuantParam`.

---

Could you share any experimental results on #13480? It would be great to show the impact of your work to the public :)

Reply:
I will try to make the changes as you suggested. Thanks :D
Here's what I shared about my experimental results.

https://github.com/Samsung/ONE/issues/13480#issuecomment-2270215801

Reply:
@jinevening , I've reflected the suggested changes in the [Draft PR](https://github.com/Samsung/ONE/pull/13585). If the changes look good, I'll proceed to write unit tests.